### PR TITLE
Update parse of NPCs, now we can use 'directions constants'.

### PR DESCRIPTION
**Pull request by Cretino**
eg:
prontera,155,168,DIR_WEST	script	Testing#1	4W_F_KAFRA2,{}

Old way still working. :)

Signed-off-by: Cretino <yuri-lolz@hotmail.com>

### CODE REVIEW

**Comment by MishimaHaruna** [comment]
Nice :D I wanted this for a while, those 0-7 numbers are so confusing

Could you please edit the commits to follow the [coding style guidelines](https://github.com/HerculesWS/Hercules/wiki/Coding-Style) before I start reviewing the code?

**Comment by Cretino** [comment]
Ok, I'll see how do make it. :)

**Comment by Cretino** [comment]
I was reading the wiki, but I couldn't understand so much, (my English is not very good) you can help me and see if everything is right now?

File: src/map/npc.c [hunk 1]
```diff
@@ -3012,6 +3023,19 @@ const char *npc_parse_script(const char *w1, const char *w2, const char *w3, con
 	script_start = strstr(start,",{");
 	end = strchr(start,'\n');
 
+	// Skipping floating npcs
+	if ( m != -1 ) {
```
Comment (dastgir) [inline src/map/npc.c]: should be 
```c
if (m != -1) {
```
no spaces between parenthesis and conditions

File: src/map/npc.c [hunk 2]
```diff
@@ -3317,6 +3342,17 @@ const char *npc_parse_duplicate(const char *w1, const char *w2, const char *w3,
 			if (retval) *retval = EXIT_FAILURE;
 			return end;// next line, try to continue
 		}
+
+		// Using 'npc->viewisid' to checks if given 'dir_name' is an interger or constant. [Cretino]
+		if (!npc->viewisid(dir_name)) {
+			if (!script->get_constant(dir_name, &dir)) {
+				ShowError("npc_parse_duplicate: Invalid facing constant '%s' in file '%s', line '%d'.\n", dir_name, filepath, strline(buffer,start-buffer));
+				if (retval) *retval = EXIT_FAILURE;
```
Comment (dastgir) [inline src/map/npc.c]: try to split in different lines
```c
if (retval)
    *retval = EXIT_FAILURE;
```

File: src/map/npc.c [hunk 3]
```diff
@@ -3012,6 +3023,19 @@ const char *npc_parse_script(const char *w1, const char *w2, const char *w3, con
 	script_start = strstr(start,",{");
 	end = strchr(start,'\n');
 
+	// Skipping floating npcs
+	if ( m != -1 ) {
+		// Using 'npc->viewisid' to checks if given 'dir_name' is an interger or constant. [Cretino]
+		if (!npc->viewisid(dir_name)) {
+			if (!script->get_constant(dir_name, &dir)) {
+				ShowError("npc_parse_script: Invalid facing constant '%s' in file '%s', line '%d'.\n", dir_name, filepath, strline(buffer,start-buffer));
+				if (retval) *retval = EXIT_FAILURE;
```
Comment (dastgir) [inline src/map/npc.c]: try to split in different lines
```c
if (retval)
    *retval = EXIT_FAILURE;
```

File: src/map/npc.c [hunk 4]
```diff
@@ -2759,20 +2760,29 @@ const char *npc_parse_shop(const char *w1, const char *w2, const char *w3, const
...
 			if (retval) *retval = EXIT_FAILURE;
-			return strchr(start,'\n');// skip and continue
+			return strchr(start,'\n');//continue
 		}
 
+		// Using 'npc->viewisid' to checks if given 'dir_name' is an interger or constant. [Cretino]
+		if (!npc->viewisid(dir_name)) {
+			if (!script->get_constant(dir_name, &dir)) {
+				ShowError("npc_parse_shop: Invalid facing constant '%s' in file '%s', line '%d'.\n", dir_name, filepath, strline(buffer,start-buffer));
+				if (retval) *retval = EXIT_FAILURE;
```
Comment (dastgir) [inline src/map/npc.c]: try to split in different lines
```c
if (retval)
    *retval = EXIT_FAILURE;
```

**Comment by Cretino** [comment]
Thanks @dastgir, I'm a little confused, now I understand more.

**Pull request closed without merging**